cli/tools/check.rs

```diff
@@ -34,12 +46,156 @@ use crate::tsc;
 use crate::tsc::Diagnostics;
 use crate::tsc::TypeCheckingCjsTracker;
 use crate::util::extract;
+use crate::util::fs::collect_specifiers;
+use crate::util::path::is_script_ext;
 use crate::util::path::to_percent_decoded_str;
 
 pub async fn check(
   flags: Arc<Flags>,
   check_flags: CheckFlags,
 ) -> Result<(), AnyError> {
+  let is_discovered_config = match flags.config_flag {
+    ConfigFlag::Discover => true,
+    ConfigFlag::Path(_) => false,
+    ConfigFlag::Disabled => false,
+  };
+  if is_discovered_config {
```

Comment:
Why do we only do this for discovered configuration files? Can you add a comment explaining?